sdk/src/types/block/macro.rs

```diff
@@ -49,6 +49,23 @@ macro_rules! impl_id {
             }
         }
 
+        #[cfg(feature="std")]
```

Comment:
This shouldn't be required?

Reply:
do you refer to the feature flag? The `String` belongs to the std and requires mem allocation. Without it, the no_std CI check fails.

Reply:
It requires `alloc` not `std.

Reply:
yeah, alloc is reexported by std, so I've made the direct reference on `alloc` only.